src/components/PrettyTx/PrettyTx.component.tsx

```diff
@@ -1,31 +1,85 @@
+import { useState } from 'react';
 import Button from '@mui/material/Button';
 import Tooltip from '@mui/material/Tooltip';
 import Typography from '@mui/material/Typography';
 
-import { prettyTx } from '../../utils/helpers';
-
+import IconButton from '@mui/material/IconButton';
+import contentCopy from '../../assets/icons/content-copy.svg';
+import contentCopyDone from '../../assets/icons/done.svg';
 import { PrettyTxProps } from './PrettyTx.types';
+import { prettyTx } from '../../utils/helpers';
+import { useUpdatePrettyTxUrl } from './PrettyTx.hooks';
 
 export const PrettyTx = ({
   value,
   color = 'primary',
   variant = 'body2',
-}: PrettyTxProps) => (
-  <Tooltip
-    arrow
-    title={
-      <Typography variant="body2" color={color}>
-        {value}
-      </Typography>
+}: PrettyTxProps) => {
+  const [isCopyClicked, setClickActive] = useState(false);
+  const iconData = {
```

Comment:
Move it outside the component so it won't be created every time when component rerenders